### PR TITLE
daniel/eng-310 EventCancelledError specificity

### PR DESCRIPTION
**Pull request by DanielHHowell**
Changes ServoStatus from generic failed to cancelled (unique from 'aborted' which is utilized in the upcoming fast-fail feature)

### CODE REVIEW

File: servo/api.py [hunk 1]
```diff
@@ -188,7 +191,7 @@ async def report_progress(self, **kwargs) -> None:
             raise servo.errors.UnexpectedEventError(status.reason)
         elif status.status == OptimizerStatuses.cancelled:
             # Optimizer wants to cancel the operation
-            raise servo.errors.EventCancelledError(status.reason)
+            raise servo.errors.EventCancelledError("Command cancelled")
```
Comment (linkous8) [inline servo/api.py:194]: Please update this to be `(status.reason or "Command cancelled")` so the backend status is communicated when present

**Review by linkous8 — CHANGES_REQUESTED** [review]
Please expand the unit test here to test EventCancelledError: https://github.com/opsani/servox/blob/9ad1d2f69cd605f24d0a2bacbe60fe1835174311/tests/api_test.py#L7

File: servo/api.py (hunk 1 above)
Comment (DanielHHowell) [reply servo/api.py]: Good call, thank you

**Review by linkous8 — APPROVED** [review]
LGTM

**Pull request merged**